check.cpp (checker) file not compiling because of testlib.h

## check.cpp (checker) file not compiling because of testlib.h
**Description:** I get this error whenever I try to import tasks using **cmsImportTask**:

**Steps to reproduce:**
- installed CMS 1.4-rc0
- extract polygon problem
- launch command: cmsImportTask -L polygon_task -c 1 moi-problem-math

But I get the following output:

<pre>In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4495:8: error: redefinition of ‘struct is_iterable<T>’
 struct is_iterable
        ^~~~~~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4558:8: note: previous definition of ‘struct is_iterable<T>’
 struct is_iterable
        ^~~~~~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4507:8: error: redefinition of default argument for ‘class T’
 struct __testlib_enable_if {};
        ^~~~~~~~~~~~~~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4569:18: note: original definition appeared here
 template<bool B, class T = void>
                  ^~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4510:8: error: redefinition of ‘struct __testlib_enable_if<true, T>’
 struct __testlib_enable_if<true, T> { typedef T type; };
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4573:8: note: previous definition of ‘struct __testlib_enable_if<true, T>’
 struct __testlib_enable_if<true, T> { typedef T type; };
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4513:66: error: redefinition of ‘template<class T> typename __testlib_enable_if<(! is_iterable<T>::value), void>::type __testlib_print_one(const T&)’
 typename __testlib_enable_if<!is_iterable<T>::value, void>::type __testlib_print_one(const T& t)
                                                                  ^~~~~~~~~~~~~~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4576:66: note: ‘template<class T> typename __testlib_enable_if<(! is_iterable<T>::value), void>::type __testlib_print_one(const T&)’ previously declared here
 typename __testlib_enable_if<!is_iterable<T>::value, void>::type __testlib_print_one(const T& t)
                                                                  ^~~~~~~~~~~~~~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4519:65: error: redefinition of ‘template<class T> typename __testlib_enable_if<is_iterable<T>::value, void>::type __testlib_print_one(const T&)’
 typename __testlib_enable_if<is_iterable<T>::value, void>::type __testlib_print_one(const T& t)
                                                                 ^~~~~~~~~~~~~~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4582:65: note: ‘template<class T> typename __testlib_enable_if<is_iterable<T>::value, void>::type __testlib_print_one(const T&)’ previously declared here
 typename __testlib_enable_if<is_iterable<T>::value, void>::type __testlib_print_one(const T& t)
                                                                 ^~~~~~~~~~~~~~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4533:75: error: redefinition of ‘typename __testlib_enable_if<is_iterable<T>::value, void>::type __testlib_print_one(const T&) [with T = std::__cxx11::basic_string<char>; typename __testlib_enable_if<is_iterable<T>::value, void>::type = void]’
 typename __testlib_enable_if<is_iterable<std::string>::value, void>::type __testlib_print_one<std::string>(const std::string& t)
                                                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4596:75: note: ‘typename __testlib_enable_if<is_iterable<T>::value, void>::type __testlib_print_one(const T&) [with T = std::__cxx11::basic_string<char>; typename __testlib_enable_if<is_iterable<T>::value, void>::type = void]’ previously declared here
 typename __testlib_enable_if<is_iterable<std::string>::value, void>::type __testlib_print_one<std::string>(const std::string& t)
                                                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4539:6: error: redefinition of ‘template<class A, class B> void __println_range(A, B)’
 void __println_range(A begin, B end)
      ^~~~~~~~~~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4602:6: note: ‘template<class A, class B> void __println_range(A, B)’ previously declared here
 void __println_range(A begin, B end)
      ^~~~~~~~~~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4554:8: error: redefinition of default argument for ‘class Enable’
 struct is_iterator
        ^~~~~~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4616:19: note: original definition appeared here
 template<class T, class Enable = void>
                   ^~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4565:8: error: redefinition of ‘struct is_iterator<T, typename __testlib_enable_if<std::is_array< <template-parameter-1-1> >::value>::type>’
 struct is_iterator<T, typename __testlib_enable_if<std::is_array<T>::value >::type>
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4628:8: note: previous definition of ‘struct is_iterator<T, typename __testlib_enable_if<std::is_array< <template-parameter-1-1> >::value>::type>’
 struct is_iterator<T, typename __testlib_enable_if<std::is_array<T>::value >::type>
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4571:66: error: redefinition of ‘template<class A, class B> typename __testlib_enable_if<(! is_iterator<B>::value), void>::type println(const A&, const B&)’
 typename __testlib_enable_if<!is_iterator<B>::value, void>::type println(const A& a, const B& b)
                                                                  ^~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4634:66: note: ‘template<class A, class B> typename __testlib_enable_if<(! is_iterator<B>::value), void>::type println(const A&, const B&)’ previously declared here
 typename __testlib_enable_if<!is_iterator<B>::value, void>::type println(const A& a, const B& b)
                                                                  ^~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4580:65: error: redefinition of ‘template<class A, class B> typename __testlib_enable_if<is_iterator<B>::value, void>::type println(const A&, const B&)’
 typename __testlib_enable_if<is_iterator<B>::value, void>::type println(const A& a, const B& b)
                                                                 ^~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4643:65: note: ‘template<class A, class B> typename __testlib_enable_if<is_iterator<B>::value, void>::type println(const A&, const B&)’ previously declared here
 typename __testlib_enable_if<is_iterator<B>::value, void>::type println(const A& a, const B& b)
                                                                 ^~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4586:6: error: redefinition of ‘template<class A> void println(const A*, const A*)’
 void println(const A* a, const A* b)
      ^~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4649:6: note: ‘template<class A> void println(const A*, const A*)’ previously declared here
 void println(const A* a, const A* b)
      ^~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4592:6: error: redefinition of ‘void println(const A*, const A*) [with A = char]’
 void println<char>(const char* a, const char* b)
      ^~~~~~~~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4655:6: note: ‘void println(const A*, const A*) [with A = char]’ previously declared here
 void println<char>(const char* a, const char* b)
      ^~~~~~~~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4601:6: error: redefinition of ‘template<class T> void println(const T&)’
 void println(const T& x)
      ^~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4664:6: note: ‘template<class T> void println(const T&)’ previously declared here
 void println(const T& x)
      ^~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4608:6: error: redefinition of ‘template<class A, class B, class C> void println(const A&, const B&, const C&)’
 void println(const A& a, const B& b, const C& c)
      ^~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4671:6: note: ‘template<class A, class B, class C> void println(const A&, const B&, const C&)’ previously declared here
 void println(const A& a, const B& b, const C& c)
      ^~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4619:6: error: redefinition of ‘template<class A, class B, class C, class D> void println(const A&, const B&, const C&, const D&)’
 void println(const A& a, const B& b, const C& c, const D& d)
      ^~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4682:6: note: ‘template<class A, class B, class C, class D> void println(const A&, const B&, const C&, const D&)’ previously declared here
 void println(const A& a, const B& b, const C& c, const D& d)
      ^~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4632:6: error: redefinition of ‘template<class A, class B, class C, class D, class E> void println(const A&, const B&, const C&, const D&, const E&)’
 void println(const A& a, const B& b, const C& c, const D& d, const E& e)
      ^~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4695:6: note: ‘template<class A, class B, class C, class D, class E> void println(const A&, const B&, const C&, const D&, const E&)’ previously declared here
 void println(const A& a, const B& b, const C& c, const D& d, const E& e)
      ^~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4647:6: error: redefinition of ‘template<class A, class B, class C, class D, class E, class F> void println(const A&, const B&, const C&, const D&, const E&, const F&)’
 void println(const A& a, const B& b, const C& c, const D& d, const E& e, const F& f)
      ^~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4710:6: note: ‘template<class A, class B, class C, class D, class E, class F> void println(const A&, const B&, const C&, const D&, const E&, const F&)’ previously declared here
 void println(const A& a, const B& b, const C& c, const D& d, const E& e, const F& f)
      ^~~~~~~
In file included from /home/mouad/moi-problem-math/files/check.cpp:1:0:
/home/mouad/moi-problem-math/files/testlib.h:4664:6: error: redefinition of ‘template<class A, class B, class C, class D, class E, class F, class G> void println(const A&, const B&, const C&, const D&, const E&, const F&, const G&)’
 void println(const A& a, const B& b, const C& c, const D& d, const E& e, const F& f, const G& g)
      ^~~~~~~
In file included from <command-line>:0:0:
/usr/local/include/cms/testlib.h:4727:6: note: ‘template<class A, class B, class C, class D, class E, class F, class G> void println(const A&, const B&, const C&, const D&, const E&, const F&, const G&)’ previously declared here
 void println(const A& a, const B& b, const C& c, const D& d, const E& e, const F& f, const G& g)
      ^~~~~~~
2018-10-27 20:21:32,361 - CRITICAL [<unknown>] Could not compile checker
</pre>


### System Information

**CMS version:** 1.4.rc-0
**Was CMS installed:** yes
**Using a virtualenv:** no
**Distribution and version:** Ubuntu 18.04

--- a/cmscontrib/loaders/polygon/testlib.h
+++ b/cmscontrib/loaders/polygon/testlib.h
@@ -55,6 +55,9 @@
  *   - Add the "CMS_VERBOSE_FEEDBACK" conditional macro to enable
  *     testlib-style messages
  *   - Interactors are not supported
+ *
+ * Backports:
+ *   - 1bcfacc3a97667b38a3aef9d95b97edc6a9db688 (MikeMirzayanov/testlib#79)
  */
 
 /* NOTE: This file contains testlib library for C++.
@@ -79,6 +82,7 @@
  */
 
 const char* latestFeatures[] = {
+                          "Fixed issue #79: fixed missed guard against repeated header include",
                           "Fixed stringstream repeated usage issue",
                           "Fixed compilation in g++ (for std=c++03)",
                           "Batch of println functions (support collections, iterator ranges)",
@@ -4564,8 +4568,6 @@ NORETURN void expectedButFound<long double>(TResult result, long double expected
     __testlib_expectedButFound(result, double(expected), double(found), prepend.c_str());
 }
 
-#endif
-
 #if __cplusplus > 199711L || defined(_MSC_VER)
 template <typename T>
 struct is_iterable
@@ -4754,4 +4756,5 @@ void println(const A& a, const B& b, const C& c, const D& d, const E& e, const F
     __testlib_print_one(g);
     std::cout << std::endl;
 }
+#endif
 #endif

